raiden-ts/src/transport/epics/presence.ts

```diff
@@ -156,40 +157,39 @@ export function matrixMonitorChannelPresenceEpic(
 }
 
 /**
- * Update our matrix's avatarUrl on config.caps changes
+ * Update our matrix's avatarUrl on config.caps on startup and changes
  *
  * @param action$ - Observable of RaidenActions
  * @param state$ - Observable of RaidenStates
  * @param deps - Epics dependencies
  * @param deps.matrix$ - MatrixClient async subject
  * @param deps.config$ - Config object
+ * @param deps.init$ - Init$ subject
  * @returns Observable which never emits
  */
 export function matrixUpdateCapsEpic(
   action$: Observable<RaidenAction>,
   {}: Observable<RaidenState>,
-  { matrix$, config$ }: RaidenEpicDeps,
+  { matrix$, config$, init$ }: RaidenEpicDeps,
 ): Observable<never> {
   return config$.pipe(
     completeWith(action$),
     pluck('caps'),
     distinctUntilChanged(isEqual),
-    skip(1), // skip replay(1) and act only on changes
-    switchMap((caps) =>
+    withLatestFrom(init$.pipe(ignoreElements(), startWith(false), endWith(true))),
```

Comment:
Am I correct that this is the feature to wait until the initialization is done
before we start this epic? I'm a little confused how this works combined with
the `filter(() => synced)` below. So why do we filter this just after we set the
avatar URL?

Reply:
This doesn't wait until it is synced to start the epic, it is started from start and will act as soon as `matrix$` is set. This `withLatestFrom` will set `synced` as false on the updates while Raiden is still syncing (including initial set) and as true afterwards. The filter after the `setAvatarUrl` will make it be set in both cases, but toggling the presence will only happen when already synced, because before that we're still not _online_, the filter will skip the laters.

Reply:
Okay, makes sense.